src/main/shared/types/permissions.ts

```diff
@@ -0,0 +1,58 @@
+export interface PermissionMeta {
+    /**
+     * The time the permission was granted.
+     * @type {number}
+     */
+    startAt: number;
+
+    /**
+     * The duration of the permission.
+     * @type {number}
+     */
+    duration: number;
+}
+
+export interface PermissionsDocumentMixin {
+    /**
+     * The permissions associated with this document.
+     * @type {string[]}
+     */
+    permissions?: string[];
+
+    /**
+     * The permissions metadata associated with this document.
+     * @type {Record<string, { startAt: number; duration: number }>}
+     */
+    permissionsMeta?: Record<string, PermissionMeta>;
+}
+
+
+export interface GroupsDocumentMixin {
+    /**
+     * The groups associated with this document.
+     * @type {string[]}
+     */
+    groups?: string[];
+}
+
+export interface PermissionOptions {
+    /**
+     * The permissions that are allowed to lock/unlock the door.
+     *
+     * @type {Record<'account' | 'character', string[]>}
+     */
+    permissions: {
+        account: string[];
+        character: string[];
+    };
+
+    /**
+     * The groups that are allowed to lock/unlock the door.
+     *
+     * @type {Record<'account' | 'character', string[]>}
+     */
+    groups: {
```

Comment:
Still not sure about having this, according to new system.

Example:
`group_1` has: `['foo', 'bar']` permissions.
Character A has groups: `['group_1']`
I'm checking if Character A has a `foo` permission - yes he has, because he is member of a group that has this permission.

So, maybe, we can get rid of that and just use permissions to check for commands/doors?